content/posts/2020-07-15-tweeting-from-pull-requests/index.md

```diff
@@ -0,0 +1,47 @@
+
+However, one of the biggest concerns I had when we started the idea of sharing our work was to not let our account be managed by a single person or a small group in the company. Having specific people responsible for creating those tweets would likely create bottlenecks that could lead to lack of engagement in generating new content. And to make this initiative successful, we need to make sure content creation comes from everyone. It should be accessible to anyone willing to contribute to it. I believe this is what creates a collective sense of ownership.
+
+## The problem
+
+Twitter is a [microblogging](https://en.wikipedia.org/wiki/Microblogging) platform so I thought: "Why not having a similar experience from regular blogs where multiple authors write multiple articles?". Well, that would be nice. But unfortunately, different from regular blogs, we can only post content from a single account.
+
+The easiest solution would be sharing the credentials, some may think. That would work but that only wouldn’t be enough. Collective ownership is also about making sure decisions are made and agreed upon. We would need at least people reviewing each other’s content.
+
+## The solution
+
+What I envisioned was a flow that would allow us to:
+
+1. have someone proposing a new tweet
+2. notify others when there’s new content to be reviewed
+3. have others reviewing and approving it
+4. publish tweet when content is approved
+
+![./diagram.png](diagram.png)
+
+Because these reviewers are mostly engineers, the closer this whole process is from their natural habitat, the better. We use GitHub and Slack all day so the solution would definitely be somewhere in there.
+
+I have used [Danger](https://danger.systems/js/) and [Peril](https://github.com/danger/peril) quite a lot in the past. It’s a great tool to automate many of the workflows we have, especially those involving review of Pull Requests. In fact, Loadsmart has its own set of [settings](https://github.com/loadsmart/peril-settings) that are used across many of our repositories on GitHub. But this time I decided to give GitHub actions a try.
+
+Surprisingly, most of the things I was planning to implement were already done by [Gregor Martynus](https://github.com/gr2m) in their [twitter-together](https://github.com/gr2m/twitter-together) project. From the steps of the diagram above, the Action takes care of 1, 2, and 4. The only step missing here is the Slack notification.
+
+There are a few Actions to interact with Slack on GitHub’s marketplace. They are all very similar to each other and most of them are not as flexible as I wanted when it comes to customization of the content for the notification. You can have this full flexibility by using Slack’s [Block Kit API](https://api.slack.com/block-kit) though, but apparently it would not be possible if I have decided to use one of the available Actions. Also, the fact that there’s no concept of "official" Actions made me think twice.
```

Comment:
but apparently it is not possible if I have already decided to use one of the available Actions.